sdk/eventgrid/eventgrid/review/eventgrid.api.md

```diff
@@ -0,0 +1,206 @@
+## API Report File for "@azure/eventgrid"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AzureKeyCredential } from '@azure/core-auth';
+import { KeyCredential } from '@azure/core-auth';
+import { OperationOptions } from '@azure/core-http';
+import { PipelineOptions } from '@azure/core-http';
+import { RestResponse } from '@azure/core-http';
+
+export { AzureKeyCredential }
+
+// @public
+export interface CloudEvent<T> {
+    [keyof: string]: any;
+    data?: T;
+    datacontenttype?: string;
```

Comment:
I'm surprised the name is all lower case rather than camel

Reply:
I agree this is a little weird.  The CloudEvent's specification lays out the required properties of a cloud event, and all the names in the specification are [lowercase](https://github.com/cloudevents/spec/blob/v1.0/spec.md#required-attributes) instead of camel case.  In addition, the JSON encoding of these events also uses [lower case for all the property names](https://github.com/cloudevents/spec/blob/v1.0/json-format.md).

Logically, this `CloudEvent<T>` type represents the structure outlined in the core specification (and the unexported `CloudEvent` interface in the generated model represents JSON encoding, since that's what we sent over the wire).  I decided to keep the casing from the cloud events specification, since folks who have used cloudevents before would be familiar with that name, but I'm happy to make them camel case if we prefer.

For what it's worth, the [SDK for working with cloud events published by the CNCF for TypeScript](https://github.com/cloudevents/sdk-javascript) uses lowercase property names as well.

Reply:
Having an indexer on such a key type is kind of risky because typos aren't captured. E.g. `const e: CloudEvent<number> = { date: 1}` is fine, no compilation errors. Is it possible to avoid this indexer at all?

Reply:
@bterlson, yes we can.  The goal of this indexer is to allow arbitrary properties which are placed on the envelope (the CloudEvent spec calls these Extension Context Attributes).  We could instead have a `extensions` property of type `Record<string, unknown>`.  I'll push that through.